File: cs161a/a02.cpp
#include <iostream>
#include <iomanip>
#include <cmath>
using namespace std;

int main() {

  const double LOAD_FACTOR = 0.64;
  const double PI = M_PI;

  double gumballRadius;
  double containerSize;
  double gumballVolume;
  int gumballCount;

  cout << "Welcome to my gumball guesser program!" << endl;
  cout << "Enter the radius of a gumball (cm) and the volume of a jar (mL) separated by a space: ";

  cin >> gumballRadius >> containerSize;
  while (cin >> gumballRadius >> containerSize && containerSize <= 0) {
    cout << "Enter the radius of a gumball (cm) and the volume of a jar (mL) separated by a space: ";
  }

  gumballVolume = (4.0 / 3.0) * M_PI * pow(gumballRadius, 3.0);
  gumballCount = (containerSize * LOAD_FACTOR ) / gumballVolume;

  cout << "Estimate of gumballs in the jar: " << fixed << setprecision(2) << gumballCount << endl;
  cout << "Thank you for using my program!";

}

File: cs161a_python/a02.py
#!/bin/bash python

import sys
import math

LOAD_FACTOR = 0.64
PI = math.pi


def main():

  print("Welcome to my gumball guesser program!\n")
  gumballRadius = float(input("Enter the radius of a gumball: "))
  containerSize = float(input("Enter the volume of your jar: "))
  gumballVolume = (4.0 / 3.0) * PI * gumballRadius**3
  gumballCount = round(((containerSize * LOAD_FACTOR) / gumballVolume),2);

  print("Estimate of gumballs in the jar: {}".format(gumballCount))
  print("Thank you for using my program")

    

if __name__ == "__main__":
  main()
